capa/features/extractors/elf.py

```diff
@@ -91,6 +91,24 @@ class Shdr:
     entsize: int
     buf: bytes
 
+    @classmethod
+    def from_viv(cls, section, buf: bytes):
+        """
+        construct a Shdr object from vivisect's representation of
+        section headers (Elf.Elf32Section or Elf.Elf64Section)
+        """
+        return cls(
+            int(section.vsGetField("sh_name")),
+            int(section.vsGetField("sh_type")),
+            int(section.vsGetField("sh_flags")),
+            int(section.vsGetField("sh_addr")),
+            int(section.vsGetField("sh_offset")),
+            int(section.vsGetField("sh_size")),
+            int(section.vsGetField("sh_link")),
+            int(section.vsGetField("sh_entsize")),
```

Comment:
```suggestion
            sectionsh_name,
            section.sh_type,
            section.sh_flags,
            section.sh_addr,
            section.sh_offset,
            section.sh_size,
            section.sh_link,
            section.sh_entsize,
```

I think you can access the properties of a vstruct directly.

Reply:
I wasn't aware of that, thank you!